server/src/test/java/org/opensearch/repositories/blobstore/BlobStoreRepositoryRemoteIndexTests.java

```diff
@@ -368,4 +378,239 @@ public void testRetrieveShallowCopySnapshotCase2() throws IOException {
+
+        logger.info("--> creating a remote store enabled index and indexing documents");
+        final String remoteStoreIndexName = "test-rs-idx";
+        Settings indexSettings = getRemoteStoreBackedIndexSettings(remoteStoreRepositoryName);
+        createIndex(remoteStoreIndexName, indexSettings);
+        indexDocuments(client, remoteStoreIndexName);
+
+        logger.info("--> create first shallow snapshot");
+        CreateSnapshotResponse createSnapshotResponse = client.admin()
+            .cluster()
+            .prepareCreateSnapshot(snapshotRepositoryName, "test-snap-1")
+            .setWaitForCompletion(true)
+            .setIndices(indexName, remoteStoreIndexName)
+            .get();
+        final SnapshotId snapshotId1 = createSnapshotResponse.getSnapshotInfo().snapshotId();
+
+        String[] lockFiles = getLockFilesInRemoteStore(remoteStoreIndexName, remoteStoreRepositoryName);
+        assert (lockFiles.length == 1) : "there should be only one lock file, but found " + Arrays.toString(lockFiles);
+        assert lockFiles[0].endsWith(snapshotId1.getUUID() + ".lock");
+
+        Settings snapshotRepoSettingsForShallowCopy = Settings.builder()
+            .put(snapshotRepoSettings)
+            .put(BlobStoreRepository.REMOTE_STORE_INDEX_SHALLOW_COPY.getKey(), Boolean.FALSE)
+            .build();
+        updateRepository(client, snapshotRepositoryName, snapshotRepoSettingsForShallowCopy);
+
+        logger.info("--> create full copy snapshot");
+        createSnapshotResponse = client.admin()
+            .cluster()
+            .prepareCreateSnapshot(snapshotRepositoryName, "test-snap-2")
+            .setWaitForCompletion(true)
+            .setIndices(indexName, remoteStoreIndexName)
+            .get();
+        final SnapshotId snapshotId2 = createSnapshotResponse.getSnapshotInfo().snapshotId();
+
+        lockFiles = getLockFilesInRemoteStore(remoteStoreIndexName, remoteStoreRepositoryName);
+        assert (lockFiles.length == 1) : "there should be only one lock file, but found " + Arrays.toString(lockFiles);
+
+        logger.info("--> delete full copy snapshot");
+        assertAcked(client().admin().cluster().prepareDeleteSnapshot(snapshotRepositoryName, "test-snap-2").get());
+
+        logger.info("--> create another full copy snapshot");
+        createSnapshotResponse = client.admin()
+            .cluster()
+            .prepareCreateSnapshot(snapshotRepositoryName, "test-snap-3")
+            .setWaitForCompletion(true)
+            .setIndices(indexName, remoteStoreIndexName)
+            .get();
+        final SnapshotId snapshotId3 = createSnapshotResponse.getSnapshotInfo().snapshotId();
+
+        lockFiles = getLockFilesInRemoteStore(remoteStoreIndexName, remoteStoreRepositoryName);
+        assert (lockFiles.length == 1) : "there should be only one lock file, but found " + Arrays.toString(lockFiles);
+        assert lockFiles[0].endsWith(snapshotId1.getUUID() + ".lock");
```

Comment:
use `assertEquals` or `assertTrue` like semantics in tests instead of `assert` everywhere